src/Voron/Impl/FileHeaders/FileHeader.cs

```diff
@@ -1,14 +1,20 @@
-﻿using System.Runtime.InteropServices;
+﻿﻿using System;
+ using System.Runtime.InteropServices;
 using Voron.Data.BTrees;
 using Voron.Impl.Backup;
 using Voron.Impl.Journal;
 
 namespace Voron.Impl.FileHeaders
 {
     [StructLayout(LayoutKind.Explicit, Pack = 1)]
-    public struct FileHeader
+    public unsafe struct FileHeader
     {
-        public static int HashOffset = (int)Marshal.OffsetOf<FileHeader>(nameof(Hash));
+        /// <summary>
+        /// This is _always_ the last 8 bytes
+        /// </summary>
+        public static int HashOffset = sizeof(FileHeader) - sizeof(ulong);
+        
+        public static int TransactionIdOffset = (int)Marshal.OffsetOf<FileHeader>(nameof(TransactionId));
 
```

Comment:
These 2 can be marked static. By doing so we ensure that the JIT will inline them as constants.

Reply:
Yes.